File: Celebratorica/app/src/main/java/io/blushine/rmw/item/ItemEvent.java
package io.blushine.rmw.item;

import java.util.List;

import io.blushine.android.common.ObjectEvent;

/**
 * Item event
 */
class ItemEvent extends ObjectEvent<Item> {

ItemEvent(Actions action) {
	super(action);
}

ItemEvent(Actions action, Item item) {
	super(action, item);
}

ItemEvent(Actions action, List<Item> items) {
	super(action, items);
}
}

File: remember-my-wins/app/src/main/java/io/blushine/rmw/item/ItemEvent.kt
package io.blushine.rmw.item

import io.blushine.android.common.ObjectEvent

const val GET_ALL_ITEMS = "get_all_items"

/**
 * Item event
 */
internal class ItemEvent @JvmOverloads constructor(action: Actions, items: List<Item> = listOf(), val categoryId: String = "") : ObjectEvent<Item>(action, items) {
	constructor(action: ObjectEvent.Actions, item: Item) : this(action, listOf(item))
}
